src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -1368,10 +1376,22 @@ void Convolution::prepareParams() {
                    paddingL,
                    paddingR,
                    *pAttrLocal,
-                   selected_pd->getImplementationType()};
+                   selected_pd->getImplementationType(),
+                   getParentEdgeAt(1)->getParent()->isConstant()};
 
     auto engine = getEngine();
     auto builder = [&engine](const ConvKey& key) -> executorPtr {
+        // remove the requirement on weight memory layout to let primitive
+        // report the best layout for weight to be reordered dynamically at runtime
+        auto src_dt = key.inp0->getDataType();
+        auto wdt = src_dt;
+        if (src_dt == dnnl_s8 || src_dt == dnnl_u8) {
+            wdt = memory::data_type::s8;
+        }
```

Comment:
Shouldn't we retrive the data from the `key.inp1`? Of course if we've set the proper precision on the compilation stage (see previous comment).

Reply:
because we want to reorder only once from the original constant ngraph node(see previous reply), so we have to derive the actual precision here based on oneDNN's requirement.

Reply:
Ok. That does make sense. However, on lines: https://github.com/openvinotoolkit/openvino/blob/934224b14977c1363f87d0355c316873b8e0ea2d/src/plugins/intel_cpu/src/nodes/conv.cpp#L885-L887 we are doing pretty the same check. So I would propose to keep this check in one place to remove this coupling between those two code frames. To this end we can simply extract this check to a separate method.

Reply:
OK, will change